pyingest/parsers/dubcore.py

```diff
@@ -0,0 +1,156 @@
+#!/usr/bin/env python
+
+import sys
+import json
+import codecs
+from default import BaseXmlToDictParser
+
+class NoSchemaException(Exception):
+    pass
+
+class WrongSchemaException(Exception):
+    pass
+
+class UnparseableException(Exception):
+    pass
+
+
+class DublinCoreParser(BaseXmlToDictParser):
+
+    def __init__(self):
+    # make sure we are utf-8 clean on stdout, stderr
+        self.DUBC_SCHEMA = "http://www.openarchives.org/OAI/2.0/oai_dc/"
+
+
+    def check_schema(self, r):
+        schema_spec = []
+        for s in self._array(r[u'@xmlns:oai_dc']):
+            schema_spec.append(self._text(s))
+        if len(schema_spec) == 0:
+            raise NoSchemaException("Unknown record schema.")
+        elif schema_spec[0] != self.DUBC_SCHEMA:
+            raise WrongSchemaException("Wrong schema.")
+        else:
+            pass
+
+
+    def get_tag(self, r, tag):
+        value = []
+        for s in self._array(r.get(tag,[])):
+            value.append(self._text(s))
+        if len(value) == 0:
+            value = None
+        return value
+
+
+    def resource_dict(self, fp, **kwargs):
+        d = self.xmltodict(fp, **kwargs)
+        idtag = d.get('record',{}).get('header',{}).get('identifier',{})
+        r = d.get('record',{}).get('metadata',{}).get('oai_dc:dc',{})
+        return idtag, r
+
+    def make_dubc_bibcode(self, ident):
+        return ident
+
+
+    def parse(self, fp, **kwargs):
+#   Note: this generic DC parser may not return an object that has all
+#   metadata needed for a tagged record, but not every DC record will
+#   have all of that information.
+
+        output_metadata=dict()
+
+        idtag, r = self.resource_dict(fp, **kwargs)
+        try:
+            self.check_schema(r)
+        except:
+            raise UnparseableException("Cannot parse record.")
+        else:
```

Comment:
Just a style comment: AFAIK there is no need to have the code below be in the `else` statement.  Once we throw an exception, we are guaranteed that no remaining code gets executed below.